### PR TITLE
LPS-172441 upgrade snakeyaml to 2.0

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina,

This is for ticket https://issues.liferay.com/browse/LPS-172441.
Thanks for checking!

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions found.

SF issue is not related.

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions. There are a lot of SF errors but they don't look related. I saw similar issues in another PR but rerunning SF had the same issues. I think a rebase would fix it.

**Comment by tinatian** [comment]
@jeffwu0724 Please rebase.

Thanks.
Tina

**Comment by jeffwu0724** [comment]
ci:test:relevant

**Comment by jeffwu0724** [comment]
ci:test:sf

**Comment by jeffwu0724** [comment]
retrigger the test after rebase

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLUtil.java [hunk 1]
```diff
@@ -82,7 +87,7 @@ public static OpenAPIYAML loadOpenAPIYAML(String yamlString) {
 
 		configYAMLConstructor.addTypeDescription(securityTypeDescription);
 
-		Representer representer = new Representer() {
+		Representer representer = new Representer(new DumperOptions()) {
```
Comment (tinatian) [inline modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLUtil.java:90]: @jeffwu0724 Why this one is using a different one than the other two?

**Comment by tinatian** [comment]
@jeffwu0724 Please fix typo "fir the ugrade of snakeYAML 2.0"

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLUtil.java (hunk 1 above)
Comment (jeffwu0724) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLUtil.java:90]: Hi @tinatian .

Based on the documentation, we need to have DumperOptionas as parameter while declaring Representer, please see https://bitbucket.org/snakeyaml/snakeyaml/commits/094691f9392fbf6997d05aeaa4c615a434b41120

And we need to have LoaderOptions while declaring new Construcor as parameter , please see https://bitbucket.org/snakeyaml/snakeyaml/commits/68d199a0ef93fbc02639f891d06ea994b6134a0e.

I will include these two infos in my commit message.

**Comment by jeffwu0724** [comment]
upgrade the commit messages only, didn't touch the code, the previous test result should be good enough

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLUtil.java (hunk 1 above)
Comment (tinatian) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLUtil.java:90]: @jeffwu0724 Then please separate the changes into two commits.

**Comment by jeffwu0724** [comment]
ci:test:sf

**Comment by jeffwu0724** [comment]
ci:test:relevant

**Comment by jeffwu0724** [comment]
retrigger the test after split the commit

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by jeffwu0724** [comment]
Hey @Kvale1733 , there is a portal failure about this PR, please see https://liferay.slack.com/archives/CLCD3DQLF/p1679612864476009. And it said `(Can't process class [META-INF/versions/9/module-info.class] (Unsupported class version number [53.0] (maximum 52.0, Java 1.8)))`
May I ask if the ci test catch this error, please?
Thanks!

**Comment by Kvale1733** [comment]
@jeffwu0724 Looks like portal-license-smoke-jdk8 batch would have caught this. I see it was caught in other runs from different PR's. But in this PR it failed with

```
build-common-java.compile:
    [javac] Compiling 2313 source files to /opt/dev/projects/github/liferay-portal/portal-impl/classes
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:317: error: '}' expected
    [javac] 		"3d0f3ba603147eab1ebff185058c881e0fe3b4b639b57a1d3a0948458730820eef72"
    [javac] 		                                                                      ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:318: error: illegal start of type
    [javac] 		"22cba346d242f28eafe08dd665ca15a27108b89e218695e349fe02e3a38f",
    [javac] 		                                                              ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:318: error: <identifier> expected
    [javac] 		"22cba346d242f28eafe08dd665ca15a27108b89e218695e349fe02e3a38f",
    [javac] 		                                                               ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:319: error: ';' expected
    [javac] 		"cb03be22e089038d86dd4c865f945a8e42ad3c24fafef07fc8e775fb8a75932bbfc6" +
    [javac] 		                                                                      ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:320: error: illegal start of type
    [javac] 		"5b274c7ac0a7e79288d2b4e27985edcc2cf756927bee8b932390e590c5e061ebbf9b" +
    [javac] 		^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:320: error: <identifier> expected
    [javac] 		"5b274c7ac0a7e79288d2b4e27985edcc2cf756927bee8b932390e590c5e061ebbf9b" +
    [javac] 		                                                                      ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:320: error: ';' expected
    [javac] 		"5b274c7ac0a7e79288d2b4e27985edcc2cf756927bee8b932390e590c5e061ebbf9b" +
    [javac] 		                                                                        ^
    [javac] /opt/dev/projects/github/liferay-portal/portal-impl/src/com/liferay/portal/license/validator/KeyValidator.java:321: error: illegal start of type
```
[Log](https://testray.liferay.com/reports/production/logs/2023-03/test-1-40/test-portal-acceptance-pullrequest(master)/689/portal-license-smoke-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

Can't really run it locally since it uses the release tool which I don't have access to, for building a release bundle, and it looks like the failure happens at this point during the compile of that bundle. We should have rerun ci:test:stable in this case @catarineaguiar to see what that issue was with this batch.

**Comment by Kvale1733** [comment]
Here's an example of a different run for this batch where the error was caugh:https://testray.liferay.com/reports/production/logs/2023-03/test-1-26/test-portal-acceptance-pullrequest(master)/8393/portal-license-smoke-jdk8/0/0/jenkins-console.txt.gz?authuser=0

@jeffwu0724 We can make sure this doesn't happen by verifying this batch passes in the PR you resend.

**Comment by jeffwu0724** [comment]
Thanks @Kvale1733 for this detailed answer!
If we cannot reproduce this locally, does it mean I need to send my new PR to ci and make sure all the test passed on `ci:test:stable`, please?

**Comment by jeffwu0724** [comment]
ci:test:relevant

**Pull request closed without merging**